app/controllers/avo/associations_controller.rb

```diff
@@ -194,5 +204,22 @@ def has_many_reflection?
         ActiveRecord::Reflection::HasAndBelongsToManyReflection
       ]
     end
+
+    def additional_params?
+      additional_params.keys.count >= 1
+    end
+
+    def additional_params
+      params[:fields].except("related_id")
+    end
+
+    def new_join_record_params
+      field_names = Avo::FieldsExecutionContext.new(target: @field.extra, parent: @resource).detect_fields.items.map { |field| field.name.tr(" ", "_").downcase }
+      additional_params.permit(field_names).merge({source_foreign_key => @attachment_record.id, through_foreign_key => @record.id})
+    end
+
+    def new_join_record
+      reflection.through_reflection.klass.new(new_join_record_params)
+    end
```

Comment:
Params are directly used as values. 

As example, we can see that `update_using` option is not respected:
```ruby
field :patrons, as: :has_many, through: :patronships,
      translation_key: "patrons",
      extra: -> {
        field :review, as: :text, update_using: -> { ">> #{value} <<"}
      }
```

Let's try to use the `fill_field` for each value to ensure that all field options are respected and applied.

Reply:
Sorry, didn't know about `fill_record` :eyes: . I just addressed this and pushed @Paul-Bob. Please take a look and let me know :pray: